### PR TITLE
feat!: Next release

### PR DESCRIPTION
**Pull request by elshan-eth**
- add oracle role to set the price
- refactoring repo structure and test for performance 
- fix bugs from the audit report 
- removeСUFromCC had a bug with return units from the deal. Now CLI needs to make returnComputeUnitFromDeal before removeCUFomCC. https://github.com/fluencelabs/cli/pull/897

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -23,3 +27,11 @@ stage = "https://ipc-stage.fluence.dev"
 dar = "https://ipc.dar.fluence.dev"
 kras = "https://ipc.kras.fluence.dev"
 mumbai = "https://rpc.ankr.com/polygon_mumbai"
+
+[profile.test]
+src = "test"
+out = "test-out"
```
Comment (AlcibiadesCleinias) [inline foundry.toml:33]: `out-test` pls, coz we will have out close to out-test in dir tree

File: src/core/modules/capacity/interfaces/ICapacity.sol [hunk 2]
```diff
@@ -144,6 +144,7 @@ interface ICapacity {
         uint256 snapshotEpoch;
         uint256 activeUnitCount;
         uint256 nextAdditionalActiveUnitCount;
+        mapping(uint256 => uint256) successCountByEpoch;
```
Comment (elshan-eth) [inline src/core/modules/capacity/interfaces/ICapacity.sol:147]: This a collision!!!!

File: src/core/modules/capacity/interfaces/ICapacity.sol [hunk 3]
```diff
@@ -136,6 +135,7 @@ interface ICapacity {
         uint256 filledRemainingFailedUnitsInLastEpoch;
         uint256 exitedUnitCount;
         uint256 totalSlashedCollateral;
+        mapping(bytes32 => bool) isUnitExited;
```
Comment (elshan-eth) [inline src/core/modules/capacity/interfaces/ICapacity.sol:138]: This a collision!!!!

Comment (elshan-eth) [reply src/core/modules/capacity/interfaces/ICapacity.sol:138]: They are shifting the next var slot in the Commitment structure.

**Pull request closed without merging**